collectors/mesos/agent/agent.go

```diff
@@ -50,17 +51,75 @@ type Collector struct {
 	// https://godoc.org/github.com/sirupsen/logrus#Entry
 	log *logrus.Entry
 
-	metricsChan chan producers.MetricsMessage
-	nodeInfo    collectors.NodeInfo
-	timestamp   int64
+	metricsChan       chan producers.MetricsMessage
+	nodeInfo          collectors.NodeInfo
+	timestamp         int64
+	ContainerTaskRels *ContainerTaskRels
+}
+
+// ContainerTaskRels defines the relationship between containers and tasks.
+type ContainerTaskRels struct {
+	sync.Mutex
+	rels map[string]*TaskInfo
+}
+
+// NewContainerTaskRels creates a new empty ContainerTaskRels
+func NewContainerTaskRels() *ContainerTaskRels {
+	return &ContainerTaskRels{rels: make(map[string]*TaskInfo)}
+}
+
+// Get is a utility method which handles the mutex lock and abstracts the inner
+// map in ContainerTaskRels away. If no task info is available for the supplied
+// containerID, returns nil.
+func (ctr *ContainerTaskRels) Get(containerID string) *TaskInfo {
+	ctr.Lock()
+	defer ctr.Unlock()
+	if ctr.rels != nil {
```

Comment:
Do you need this anymore if you're already constructing the type with a non-nil map, here and otherwise?

Reply:
No, I guess not. Just disguising my laziness behind a front of belt-and-braces.